File: generate_dmg_bg.py
from PIL import Image, ImageDraw, ImageFont
import math

width, height = 600, 400
image = Image.new("RGBA", (width, height))
draw = ImageDraw.Draw(image)

# Gradient #1a1a1a to #2d2d2d
color1 = (26, 26, 26)
color2 = (45, 45, 45)

for y in range(height):
    r = int(color1[0] + (color2[0] - color1[0]) * y / height)
    g = int(color1[1] + (color2[1] - color1[1]) * y / height)
    b = int(color1[2] + (color2[2] - color1[2]) * y / height)
    draw.line([(0, y), (width, y)], fill=(r, g, b))

# Try to draw an arrow in the middle
cx, cy = width//2, height//2
draw.polygon([(cx - 20, cy - 10), (cx + 10, cy - 10), (cx + 10, cy - 20), (cx + 30, cy), (cx + 10, cy + 20), (cx + 10, cy + 10), (cx - 20, cy + 10)], fill=(200, 200, 200, 255))

# Draw text "Drag to install"
try:
    font = ImageFont.truetype("/System/Library/Fonts/Helvetica.ttc", 16)
except:
    font = ImageFont.load_default()

text = "Drag to install"
text_w = draw.textlength(text, font=font)
draw.text(((width - text_w)/2, cy + 30), text, fill=(255, 255, 255, 200), font=font)

image.save("build/dmg-background.png")
print("Background generated at build/dmg-background.png")

File: generate_dmg_bg.swift
import AppKit

let width: CGFloat = 600
let height: CGFloat = 400

let image = NSImage(size: NSSize(width: width, height: height))
image.lockFocus()

let context = NSGraphicsContext.current!.cgContext
let colorSpace = CGColorSpaceCreateDeviceRGB()
let colors = [NSColor(calibratedRed: 0.1, green: 0.1, blue: 0.1, alpha: 1.0).cgColor,
              NSColor(calibratedRed: 0.18, green: 0.18, blue: 0.18, alpha: 1.0).cgColor] as CFArray
let locations: [CGFloat] = [0.0, 1.0]

if let gradient = CGGradient(colorsSpace: colorSpace, colors: colors, locations: locations) {
    context.drawLinearGradient(gradient, start: CGPoint(x: 0, y: height), end: CGPoint(x: 0, y: 0), options: [])
}

let arrowPath = NSBezierPath()
let cx = width / 2
let cy = height / 2

arrowPath.move(to: NSPoint(x: cx - 20, y: cy - 10))
arrowPath.line(to: NSPoint(x: cx + 10, y: cy - 10))
arrowPath.line(to: NSPoint(x: cx + 10, y: cy - 20))
arrowPath.line(to: NSPoint(x: cx + 30, y: cy))
arrowPath.line(to: NSPoint(x: cx + 10, y: cy + 20))
arrowPath.line(to: NSPoint(x: cx + 10, y: cy + 10))
arrowPath.line(to: NSPoint(x: cx - 20, y: cy + 10))
arrowPath.close()

NSColor(white: 0.8, alpha: 1.0).setFill()
arrowPath.fill()

let text = "Drag to install" as NSString
let font = NSFont.systemFont(ofSize: 18, weight: .medium)
let attrs: [NSAttributedString.Key: Any] = [.font: font, .foregroundColor: NSColor(white: 1.0, alpha: 0.8)]
let textSize = text.size(withAttributes: attrs)
text.draw(at: NSPoint(x: (width - textSize.width) / 2, y: cy - 40), withAttributes: attrs)

image.unlockFocus()

if let tiffData = image.tiffRepresentation,
   let bitmap = NSBitmapImageRep(data: tiffData),
   let pngData = bitmap.representation(using: .png, properties: [:]) {
    let url = URL(fileURLWithPath: "build/dmg-background.png")
    try? pngData.write(to: url)
    print("Background generated at build/dmg-background.png")
}
